bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractResourceManager.java

```diff
@@ -384,9 +384,6 @@ public synchronized W beginNodeTrx(final @Nonnegative int maxNodeCount, final @N
       final @Nonnull TimeUnit timeUnit, final @Nonnull AfterCommitState afterCommitState) {
     // Checks.
     assertAccess(getMostRecentRevisionNumber());
-    if (maxNodeCount < 0 || maxTime < 0) {
```

Comment:
Hm, I'm not sure if we should solely rely on the annotations in this case. What do you think, @JPDSousa?

Reply:
@JPDSousa I'd argue that we still need the precondition checks to fail fast: https://stackoverflow.com/questions/13484202/how-to-use-nullable-and-nonnull-annotations-more-effectively

For me it's just a documentation and I've not used static type checkers, but even if I think it's necessary to still check explicitly, don't you think?

Reply:
I see your point, and the fail-fast argument is a really strong one. I missed the detail that if we don't do this check, we will catch the invalid value too late (e.g., nodeTrxIDCounter.incrementAndGet() would have already ran). 👍 

However, let me re-emphasize that these annotations should live in the base method (declared in the interface), as that's where the "official" contract definition is. Also, I would then argue that declaring them here is redundant, as this class is not even meant to be API-visible.

Reply:
I see your point, you're right :-)